### ISSUE TITLE
Allow dots as part of environment variable name.

### ISSUE THREAD

**Issue by vic** [mefellows/sbt-dotenv#23]
I, like many other Scala programmers use [config](https://github.com/lightbend/config) and populate some values into it from environment variables, in my case using `sbt-dotenv`. One thing I noticed not working as expected are Config values that are lists, for example:

```
akka.cluster {
   seed-nodes = []
   seed-nodes = ${?SEED_NODES}
} 
```

Here, `SEED_NODES` is expected to be a list, and you can set values from your environment like:

`env SEED_NODES.0=foo SEED_NODES.1=bar sbt run`

The `config` library automatically converts those two values into the `SEED_NODES` list.

However, adding `SEED_NODES.0` as key on my `.env` file didn't work, and then looking at the source code, I found there is a [test](https://github.com/mefellows/sbt-dotenv/blob/master/src/test/scala/au/com/onegeek/sbtdotenv/SbtDotenvSpec.scala#L54) for exactly this. 

So, my question is, given that `sbt` is mostly used by scala people were `config` is a popular library, wouldn't it make sense to allow dots to be part of a variable name?

A workaround I found was to use the following config:

```
akka.cluster {
   seed-nodes = []
   seed-nodes += ${?SEED_NODES_1}
   seed-nodes += ${?SEED_NODES_2}
} 
```

and just place those `SEED_NODES_1` on my env file.

This, however is far from ideal, as it limits the number of max elements you can have in the array.

Could provide a patch if you feel like it, just wanted to know first the reason behind the test.

**Comment by vic** [comment]
Yes, sure. I'll ping back with a PR soon.

**Comment by vic** [comment]
Fixed by #25
